lms/djangoapps/courseware/grades.py

```diff
@@ -499,10 +559,13 @@ def _progress_summary(student, course, course_structure=None):
         course_structure = get_course_blocks(student, course.location)
     if not len(course_structure):
         return None
-    scorable_locations = [block_key for block_key in course_structure if possibly_scored(block_key)]
-
-    with outer_atomic():
-        scores_client = ScoresClient.create_for_locations(course.id, student.id, scorable_locations)
+    if scores_client is None:
+        if grading_context_result is None:
+            scorable_locations = [block_key for block_key in course_structure if possibly_scored(block_key)]
+        else:  # TODO: these two list are equivalent, yeah?
```

Comment:
The former will be a superset of the latter.

Former: A full list of all parent-type blocks and blocks with 'has_score' defined (i.e., assessments).  This will include the course block and chapters, as well.

Latter: A filtered down list of the former.  Only includes blocks within the "subsection" sub-tree, and only for subsections that are marked as graded.